### DISCUSSION TITLE
Time itterations in transfer

### DISCUSSION THREAD

**Discussion by philipJFpfahl** [idaholab/moose#22901] [Q&A Modules: Navier-Stokes] [answered]
Hey,

I have a master-app in which I do fluid dynamics with the INSFV module and a sub-app in which I calculate the particle concentration using FE. 
Now I transfer with MultiAppShapeEvaluationTransfer the concentration to my parent app and back. It seems like I can't get the timing to work. I need some transfer like this:

![Master_Sub](https://user-images.githubusercontent.com/95305404/206025297-94fdf178-65b8-44f1-82fc-4647e9e96cd6.png)

So I want the sub-app to run a full-time step. Then the master app should pull the concentration. Do advection with it and then finish the time step by pushing it to the sub-app. The sub-app should start a new time step with the newly pushed concentration.
But this transfer like this
```
[Transfers]
  [pull_C]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Subapp
    source_variable = C
    variable = C
    execute_on = TIMESTEP_INITIAL
  []
  [push_C]
    type = MultiAppShapeEvaluationTransfer
    fto_multi_app = Subapp
    source_variable = C
    variable = C
    execute_on = TIMESTEP_END
  []
[]
```
Seems not to work. At least the solution is non-physical.
I do the advection with 
```
  [C_time]
    type = FVTimeKernel
    variable = C
  []
  [C_advection]
    type = INSFVScalarFieldAdvection
    variable = C
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
  []
```
and I initialize C with
```
[Variables]
  [C]
    type = INSFVScalarFieldVariable
  []
[]
```
Some insights I have:
-The solution converges most of the time if I "just" pull and push C to from to the sub-app, but does not give a physical answer.
-It seems to work fine if I do only pull from the sub-app and execute_on = TIMESTEP_END. 
-The advection works fine on its own when I pull.

I am happy about any ideas and all suggestions.
All the best,
Philip

**Comment by GiudGiud** [comment, marked as answer]
Hello

This parameter seems mispelled
    fto_multi_app = Subapp

also this execute_on does not exist? It should be timestep_begin maybe?
```
[Transfers]
  [pull_C]
    execute_on = TIMESTEP_INITIAL
```

In general, for multiphysics iterations you want to execute the transfers on the same execute_on as the multiapp. Here you should have both on timestep_end. 

Any other execution schedule is for custom schemes and advanced use.
Guillaume

**Reply by philipJFpfahl** [reply]
Hey,
yes the comments are correct. 
it shoudhave been:
```
[Transfers]
  [pull_C]
    execute_on = TIMESTEP_BEGIN
```
and to_multi_app = Subapp.

I don't understand. What does MOOSE do, when the push and pull commands are at the same time? 
In what order are the apps acting on the variable?

Philip

**Reply by GiudGiud** [reply]
You may turn on verbose_multiapps = true in the Problem to see what moose is doing

It goes:
transfer to multiapp
run multiapp
transfer from multiapp

**Reply by philipJFpfahl** [reply]
Hey,
I checked and you are completly right. There is an other issue with the transfer and the kernel we discussed in [here](https://github.com/idaholab/moose/discussions/21752). I will make a separate discussion about that. 
Meanwhile, thank you a lot!
Philip

**Comment by philipJFpfahl** [comment]
Hey,
I dont know if I should make a new discussion about this, but it still fits here.
So I want to do a fluid simulation with FiniteVolumes and advect a concentration within this simulation. 
In a FiniteElement simulation, I want to scale the concentration over time. A minimal working example is shown below as master and sub.
I can combine the two scripts into one FV simulation for this simple example. This is not possible for my purposes, since the meshes will be different sizes, but it shows, how the final solution should be. The script is also shown below.

My questions:

-How do I tell moose that it should scale the concentration in one app and then advect it in the other? 
I think it should also be the same solution it both apps.

-Am I using the right transfers or should I use different ones?

-What is the correct execute_on?
if I just use TIMESTEP_END the concentration becomes zero everywhere.

here is the code:
master:
```
rho = 2000
mu = 50 
cp = 3750
k = 100

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = 2.0
    ymin = 0
    ymax = 2.0
    nx = 50
    ny = 50
  []
  [pin]
    type = ExtraNodesetGenerator
    input = gen
    new_boundary = 'pin'
    nodes = '0'
  []
[]

velocity_interp_method = 'rc'
advected_interp_method = 'average'
[GlobalParams]
  rhie_chow_user_object = 'rc'
[]

[UserObjects]
  [rc]
    type = INSFVRhieChowInterpolator
    u = u
    v = v
    pressure = pressure
  []
[]

[Variables]
  [u]
    type = INSFVVelocityVariable
  []
  [v]
    type = INSFVVelocityVariable
  []
  [pressure]
    type = INSFVPressureVariable
  []
  [lambda]
    family = SCALAR
    order = FIRST
  []
  [C]
    type = INSFVScalarFieldVariable
  []
[]


[AuxVariables]
  [U]
    order = CONSTANT
    family = MONOMIAL
    fv = true
  []
[]

[AuxKernels]
  [mag]
    type = VectorMagnitudeAux
    variable = U
    x = u
    y = v
  []
[]

[FVKernels]
  [mass]
    type = INSFVMassAdvection
    variable = pressure
    advected_interp_method = ${advected_interp_method}
    velocity_interp_method = ${velocity_interp_method}
    rho = ${rho}
  []
  [mean_zero_pressure]
    type = FVIntegralValueConstraint
    variable = pressure
    lambda = lambda
  []

  [u_time]
    type = INSFVMomentumTimeDerivative
    variable = 'u'
    rho = ${rho}
    momentum_component = 'x'
  []
  [u_advection]
    type = INSFVMomentumAdvection
    variable = u
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
    rho = ${rho}
    momentum_component = 'x'
  []
  [u_viscosity]
    type = INSFVMomentumDiffusion
    variable = u
    mu = ${mu}
    momentum_component = 'x'
  []
  [u_pressure]
    type = INSFVMomentumPressure
    variable = u
    momentum_component = 'x'
    pressure = pressure
  []

  [v_time]
    type = INSFVMomentumTimeDerivative
    variable = v
    rho = ${rho}
    momentum_component = 'y'
  []
  [v_advection]
    type = INSFVMomentumAdvection
    variable = v
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
    rho = ${rho}
    momentum_component = 'y'
  []
  [v_viscosity]
    type = INSFVMomentumDiffusion
    variable = v
    mu = ${mu}
    momentum_component = 'y'
  []
  [v_pressure]
    type = INSFVMomentumPressure
    variable = v
    momentum_component = 'y'
    pressure = pressure
  []
  [C_time]
    type = FVTimeKernel
    variable = C
  []
  [C_advection]
    type = INSFVScalarFieldAdvection
    variable = C
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
  []
[]

[FVBCs]
  [top_x]
    type = INSFVNoSlipWallBC
    variable = u
    boundary = 'top'
    function = 'lid_function'
  []

  [no_slip_x]
    type = INSFVNoSlipWallBC
    variable = u
    boundary = 'left right bottom'
    function = 0
  []

  [no_slip_y]
    type = INSFVNoSlipWallBC
    variable = v
    boundary = 'left right top bottom'
    function = 0
  []
[]

[Materials]
  [functor_constants]
    type = ADGenericFunctorMaterial
    prop_names = 'cp k'
    prop_values = '${cp} ${k}'
  []
[]

[Functions]
  [lid_function]
    type = ParsedFunction
    value = '0.5'
  []
[]


[Executioner]
  type = Transient
  solve_type = NEWTON
  # Run for 100+ timesteps to reach steady state.

  start_time = 0
  dt = 0.1
  end_time = 10

  petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -ksp_gmres_restart'
  petsc_options_value = 'asm      lu           NONZERO                   200'
  line_search = 'none'
  nl_rel_tol = 1e-6
  nl_max_its = 6
  l_max_its = 200
[]

[Outputs]
  exodus = true
[]

[MultiApps]
  [Sub]
    type = TransientMultiApp
    input_files = 'sub.i'
    execute_on = TIMESTEP_END
  []
[]

[Transfers]
   [pull_C_inital]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Sub
    source_variable = Concentration
    variable = C
    execute_on = INITIAL
  [] 
  [pull_C]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Sub
    source_variable = Concentration
    variable = C
    execute_on = TIMESTEP_BEGIN
  []
  [push_C]
    type = MultiAppProjectionTransfer
    #type = MultiAppShapeEvaluationTransfer
    to_multi_app = Sub
    source_variable = C
    variable = Concentration
    execute_on = TIMESTEP_END
  []
[]

```
sub:
```
#set mesh
[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = 2
    ymin = 0
    ymax = 2
    nx = 25
    ny = 25
  []
[]

[Variables]
  [Concentration]
  []
[]


[ICs]
  [Initial_Power]
    type = FunctionIC
    variable = 'Concentration'
    function =  C_func
  []
[]

[Functions]
  [C_func]
    type =ParsedFunction    
    value = 'exp(-((x-1)^2+(y-1)^2))'
  []
[]



[Kernels]
 [C_time]
    type = TimeDerivative
    variable = Concentration
 []
 [C_force]
    type = BodyForce
    variable = Concentration
    value = -0.1
 []
[]


[Executioner]
  type = Transient
  start_time = 0
  dt = 0.1
  end_time = 10
  solve_type = 'PJFNK'
[]

[Outputs]
  exodus = true
 print_nonlinear_converged_reason = false
 print_linear_converged_reason = false
  [csv]
    type = CSV
    time_data = true
  []
[]

```
Solution:
```
rho = 2000
mu = 50 
cp = 3750
k = 100

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = 2.0
    ymin = 0
    ymax = 2.0
    nx = 50
    ny = 50
  []
  [pin]
    type = ExtraNodesetGenerator
    input = gen
    new_boundary = 'pin'
    nodes = '0'
  []
[]

velocity_interp_method = 'rc'
advected_interp_method = 'average'
[GlobalParams]
  rhie_chow_user_object = 'rc'
[]

[UserObjects]
  [rc]
    type = INSFVRhieChowInterpolator
    u = u
    v = v
    pressure = pressure
  []
[]

[Variables]
  [u]
    type = INSFVVelocityVariable
  []
  [v]
    type = INSFVVelocityVariable
  []
  [pressure]
    type = INSFVPressureVariable
  []
  [lambda]
    family = SCALAR
    order = FIRST
  []
  [C]
    type = INSFVScalarFieldVariable
  []
[]

[AuxVariables]
  [U]
    order = CONSTANT
    family = MONOMIAL
    fv = true
  []
[]

[AuxKernels]
  [mag]
    type = VectorMagnitudeAux
    variable = U
    x = u
    y = v
  []
[]

[FVKernels]
  [mass]
    type = INSFVMassAdvection
    variable = pressure
    advected_interp_method = ${advected_interp_method}
    velocity_interp_method = ${velocity_interp_method}
    rho = ${rho}
  []
  [mean_zero_pressure]
    type = FVIntegralValueConstraint
    variable = pressure
    lambda = lambda
  []

  [u_time]
    type = INSFVMomentumTimeDerivative
    variable = 'u'
    rho = ${rho}
    momentum_component = 'x'
  []
  [u_advection]
    type = INSFVMomentumAdvection
    variable = u
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
    rho = ${rho}
    momentum_component = 'x'
  []
  [u_viscosity]
    type = INSFVMomentumDiffusion
    variable = u
    mu = ${mu}
    momentum_component = 'x'
  []
  [u_pressure]
    type = INSFVMomentumPressure
    variable = u
    momentum_component = 'x'
    pressure = pressure
  []

  [v_time]
    type = INSFVMomentumTimeDerivative
    variable = v
    rho = ${rho}
    momentum_component = 'y'
  []
  [v_advection]
    type = INSFVMomentumAdvection
    variable = v
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
    rho = ${rho}
    momentum_component = 'y'
  []
  [v_viscosity]
    type = INSFVMomentumDiffusion
    variable = v
    mu = ${mu}
    momentum_component = 'y'
  []
  [v_pressure]
    type = INSFVMomentumPressure
    variable = v
    momentum_component = 'y'
    pressure = pressure
  []

  [C_time]
    type = FVTimeKernel
    variable = C
  []
  [C_advection]
    type = INSFVScalarFieldAdvection
    variable = C
    velocity_interp_method = ${velocity_interp_method}
    advected_interp_method = ${advected_interp_method}
  []
  [C_force]
    type = FVCoupledForce
    variable = C
    v = -0.1
  []
[]

[FVBCs]
  [top_x]
    type = INSFVNoSlipWallBC
    variable = u
    boundary = 'top'
    function = 'lid_function'
  []

  [no_slip_x]
    type = INSFVNoSlipWallBC
    variable = u
    boundary = 'left right bottom'
    function = 0
  []

  [no_slip_y]
    type = INSFVNoSlipWallBC
    variable = v
    boundary = 'left right top bottom'
    function = 0
  []
[]

[Materials]
  [functor_constants]
    type = ADGenericFunctorMaterial
    prop_names = 'cp k'
    prop_values = '${cp} ${k}'
  []
[]

[Functions]
  [lid_function]
    type = ParsedFunction
    value = '0.5'
  []
[]


[Executioner]
  type = Transient
  solve_type = NEWTON
  # Run for 100+ timesteps to reach steady state.

  start_time = 0
  dt = 0.1
  end_time = 10

  petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -ksp_gmres_restart'
  petsc_options_value = 'asm      lu           NONZERO                   200'
  line_search = 'none'
  nl_rel_tol = 1e-6
  nl_max_its = 6
  l_max_its = 200
[]

[Outputs]
  exodus = true
[]

[MultiApps]
  [Sub]
    type = TransientMultiApp
    input_files = 'sub.i'
    execute_on = INITIAL
  []
[]

[Transfers]
   [pull_C_inital]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Sub
    source_variable = Concentration
    variable = C
    execute_on = INITIAL
  [] 
[]
```

**Comment by GiudGiud** [comment]
Hello

It doesnt seem physical to just scale a concentration on every time step.
Why is concentration the solution of two distinct PDEs ? Can you not solve them together in a single app?

Also why timestep_begin for that send?
If you do that, you will lag the quantity in the sub app solve.
Then since you are transferring the value back at the end of the time step, you will overwrite what you solved for C in the main app.

So with what you have now, you might as well not solve for C in the main app.
Maybe change this:
```
  [pull_C]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Sub
    source_variable = Concentration
    variable = C
    execute_on = TIMESTEP_BEGIN.  -> TIMESTEP_END
  []
```

Guillaume

**Reply by philipJFpfahl** [reply]
Hey, 
Again thank you for the fast answer. 
Well, the sub is just a minimal example to explain the problem. I attend to use it in a more complex coupled system with coarser meshes and different time scales.

if I change:
```
execute_on = TIMESTEP_BEGIN.  -> TIMESTEP_END
```
I "just" get a scaling of C in both apps.
Philip

**Reply by GiudGiud** [reply]
Isnt that what this is meant to do?
```
So I want to do a fluid simulation with FiniteVolumes and advect a concentration within this simulation.
In a FiniteElement simulation, I want to scale the concentration over time. A minimal working example is shown below as master and sub.
```

**Reply by philipJFpfahl** [reply]
Yes, the sub-app should scale it and the main-app should advect it.  As of now, it just gets scaled. 
I think this is because it gets pulled and pushed at the same time.

**Reply by GiudGiud** [reply]
It s not at the same time. You saw the timing of the transfers, if you set both on timestep end, you should advect, transfer, "scale" then transfer back.
This transfer back replaces the current solution post-advection. The solution is then no longer the solution of an advection problem. It is the solution of the "scaling" problem.

Another thing to consider is that you have wild non-conservative projections happening when you go from finite volume to finite element and back as the shape functions are very different.

**Reply by philipJFpfahl** [reply]
if I run it with these transfers:
```
[MultiApps]
  [Sub]
    type = TransientMultiApp
    input_files = 'sub.i'
    execute_on = TIMESTEP_END
  []
[]

[Transfers]
   [pull_C_inital]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Sub
    source_variable = Concentration
    variable = C
    execute_on = INITIAL
  [] 
  [pull_C]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = Sub
    source_variable = Concentration
    variable = C
    execute_on = TIMESTEP_END
  []
  [push_C]
    type = MultiAppProjectionTransfer
    #type = MultiAppShapeEvaluationTransfer
    to_multi_app = Sub
    source_variable = C
    variable = Concentration
    execute_on = TIMESTEP_END
  []
[]
```
It does not seem to advect and both apps just 
If I change 
```
TIMESTEP_END -> TIMESTEP_BEGIN
```
in the output, it seems like the main app advects and does not scale, and the sub-app scales, but does not advect.

**Reply by GiudGiud** [reply]
This is correct.
I dont see why the main app would scale since it does not have kernels for that.
Similarly for the subapp. 

The output of an app is the solution for the given equation. The solution of an advection problem will not look like it got scaled, even if the initial condition comes from the "scaled" solution

**Reply by philipJFpfahl** [reply]
Well, the time steps should be independent of the history. So each time step starts with an initial condition and then develops the solution over time. So I thought, that it would be possible to start from an initial condition scale it in the sub-app, send it to the master-app and use that solution as an initial condition to advect it there, send this solution back to the sub-app and then use the advected solution as an initial condition for the next time step. That is "tight coupling" in my eyes. 
Please correct me if I am wrong, I am just trying to learn!
If that is not possible, how do you do something like that? 
For me, it is important to separate the two functionalities.

**Reply by GiudGiud** [reply]
Well it only develops over a single time step in this setup. If you want to let it relax to steady state on each physics coupling iteration, a different setup will achieve that. 

This is not typical tight coupling because you are solving for the same variable in both problems. Tight coupling is usually something like  « solve one app for velocity and pressure, with a temperature dependent density, solve another app for temperature depending on the velocity field, iterate to convergence »

Maybe we should consider the approach you are taking in light of the actual physical problem you are solving. Would you mind describing it?

**Reply by philipJFpfahl** [reply]
Hey 

```
Maybe we should consider the approach you are taking in light of the actual physical problem you are solving.
```
I think that is a good idea. 
So I try to solve the lid-driven cavity reactor problem using the PK solver developed for the SEALION project within MOOSE. [Here](https://www.sciencedirect.com/science/article/pii/S0306454921004540) is the paper. It focuses on a transient benchmark for MSR.
I would like to compare the neutronics solver in this setup with other neutronics solvers. This should be within MOOSE. This is why I need a separation of the tasks.Relevant to my question is, that I do the fluid dynamics in one app and the neutronics calculation in another so that I can exchange the neutronics solver later. This is interesting for benchmarking the performance of the neutronics solver alone.
To the physics:
The concentration is in this case the concentration of the precursors. In a MSR they will be advected with the fuel and then emmit a newtron later on in the reactor at a different place. You see, they are tricky because they fit right in between the two solvers. They get advected depending on the CFD and scaled depending on the Neutronics. In the end I would like to be able to refine the number of time steps in the Neutronics app and make the mesh coarser compared to the CFD app. The chosen separation, therefore "feels" logical. 


Philip

**Reply by GiudGiud** [reply]
This input does something similar to what you want. You need to run it with NS module
https://github.com/idaholab/virtual_test_bed/blob/devel/msr/msfr/steady/run_ns.i

We will upload in the near future the CNRS benchmarks, which are lid-driven cavities with precursors.

In this problem, there is no individual scaling equation. The precursor decay happens at the same time as the advection and must be included in the same equation

**Reply by philipJFpfahl** [reply]
Are you uploading the input to [this](https://www.sciencedirect.com/science/article/pii/S0306454921004229?via%3Dihub) paper?

I know that 'we' can solve the CNRS benchmark in several different ways. I mean it has been done quite a few times now. The problem is, that it is not clear, which of the neutronics solvers performs how in which situation. 
I am interested in analyzing the CFD and the neutronics separately. To test how different neutronics solvers perform. Therefore I want to compute the CFD and the neutronics in two different apps. So that they are truly separate.
The transport of precursors is clearly a CFD problem while the production and decay are neutronics problems.

But I also understand your point, that I should apply all the kernels to a variable at once, but I think in this case it is not the best option. If I would like to do that I guess I could transfer the velocity field to the sub-app and then advect C there. This seems like more data transfer and a less accurate solution to the advection. Is there a way to alter the time stepper in MOOSE?

also: 
```
https://github.com/idaholab/virtual_test_bed/blob/devel/msr/msfr/steady/run_ns.i
```
Is super interesting too :) !

**Reply by GiudGiud** [reply]
The MSFR input is part of the paper. The cnrs is being redone in finite volume and will be uploaded soon

Neutronics solvers may be compared on how they solve the neutron transport or diffusion equation. Producing and decaying precursors is absolutely trivial there is no need to compare solvers for that. 

If you still want to do that, then you will have to transfer the velocity field to the neutronics app and solve the advection - reaction - decay equation there

**Reply by philipJFpfahl** [reply]
Oh that is great to hear. Since it appears like you send the Precursor concentration from Pronghorn to Griffin.
Or did you calculate the production of precursors from the Power profile in the Thermal Hydraulics solver? That would be a solution I can work with. Sorry for being so stubborn. I will give that change a try!

**Reply by GiudGiud** [reply]
We use the fission source distribution to compute the source in the TH solver. The fission source is transferred from the neutronics solver.

**Reply by philipJFpfahl** [reply]
That is indeed perfect for me! Thank you for your patience. I was super slow in understanding what you meant. 
Can you help me with one last thing: Is the push transfer type correct or should I use another one?
```
  [push_C]
    type = MultiAppProjectionTransfer
    to_multi_app = Sub
    source_variable = C
    variable = Concentration
    execute_on = TIMESTEP_END
  []
```

**Reply by GiudGiud** [reply]
this is fine. It s not conservative in most situations though, so you will want to use the postprocessors to be conserved arguments

**Reply by philipJFpfahl** [reply]
What do you mean by conservative? 
And which transfer should I use, if the sub-app has a coarser mesh than the main-app?

**Reply by GiudGiud** [reply]
I mean that the L1 norm of the precursors concentration will not be conserved by the transfer

**Reply by GiudGiud** [reply]
Are the subapp and main app meshes only differing by a level of uniform refinement or are they completely different meshes?

**Reply by philipJFpfahl** [reply]
They are a uniform refinement. 
Actually, I am pretty flexible about how fine the main app mesh can be.

**Reply by GiudGiud** [reply]
I've found that using geometric interpolation to go from the fine mesh to the coarse mesh works very well (almost conservative, but not quite) when you specify the appropriate number of nodes.
For example to go from a cell cut in 4 to a regular  cell, use 4 points in the origin mesh to determine the value in the target mesh. 

To go the other way, I would use shape evaluation. Depending on the FE family used, this might even be conservative (the fine mesh will reproduce the coarse solution exactly)

**Reply by philipJFpfahl** [reply]
do you do that with the num_points parameter?

Yes for the coarse -> fine mesh I am already doing that.

**Reply by GiudGiud** [reply]
yes

**Reply by philipJFpfahl** [reply]
Super thank you. 

Is there any interest in me uploading the neutronics solver?

**Reply by GiudGiud** [reply]
Uploading where? to an open source app?

**Reply by philipJFpfahl** [reply]
Yes, it also solves the CNRS benchmark within MOOSE with a new approach.

**Reply by GiudGiud** [reply]
You should definitely consider open sourcing models and code in an app. The models could go in an examples/ folder.

with regards to to the CNRS benchmarks we will also open source a solution to the full suite soon.

**Reply by philipJFpfahl** [reply]
I will consider it.
Is it going to be a deterministic solver? Might be interesting to compare the two.

**Reply by GiudGiud** [reply]
It will be griffin (deterministic). We can share the solution files, however you will need access to Griffin to run the inputs, which you can request on the NCRC website at INL

**Reply by philipJFpfahl** [reply]
The CFD part will be FV?
Yes, I am working on it. The solution might be not as interesting as the runtime and the convergence to me, so fingers crossed.
